src/factory/PlayerbotFactory.cpp

```diff
@@ -207,7 +207,10 @@ void PlayerbotFactory::Randomize(bool incremental)
     Prepare();
     LOG_DEBUG("playerbots", "Resetting player...");
     PerformanceMonitorOperation* pmo = sPerformanceMonitor->start(PERF_MON_RNDBOT, "PlayerbotFactory_Reset");
-    bot->resetTalents(true);
+    if (!sPlayerbotAIConfig->equipmentPersistence || level < sPlayerbotAIConfig->equipmentPersistenceLevel)
```

Comment:
They are functionally the same.

The condition: `if (!sPlayerbotAIConfig->equipmentPersistence || (sPlayerbotAIConfig->equipmentPersistence && level < sPlayerbotAIConfig->equipmentPersistenceLevel))` is functionally correct but redundant. 

The simplified version: `if (!sPlayerbotAIConfig->equipmentPersistence || level < sPlayerbotAIConfig->equipmentPersistenceLevel)` works the same because `!sPlayerbotAIConfig->equipmentPersistence` already covers cases where persistence is disabled, making the extra `sPlayerbotAIConfig->equipmentPersistence` check unnecessary.

I was thinking about adding `!incremental` in there too but opted against it for the moment. This is enough to solve the issues at hand.